django/test/utils.py

```diff
@@ -415,6 +418,13 @@ def enable(self):
         override = UserSettingsHolder(settings._wrapped)
         for key, new_value in self.options.items():
             setattr(override, key, new_value)
+
+        if 'SECRET_KEY' in self.options:
+            override.SECRET_KEYS = [self.options['SECRET_KEY']]
+
+        if 'SECRET_KEYS' in self.options:
+            override.SECRET_KEY = self.options['SECRET_KEYS'][0]
+
```

Comment:
Hey @apollo13 Can I get your thoughts if you have capacity please? I'm struggling to get this to work... 

I'd like to move all of this _if only one is set_ logic into `LazySettings.__getattr__()` — I've got that more or less working but `override_settings` is not playing ball. 

I'm getting the underlying settings exposed:

```
(Pdb) settings.SECRET_KEY
'django_tests_secret_key'

FAIL: test_no_secret_key (settings_tests.tests.SettingsTests)
```

I'll push where I'm at so you can see.

Reply:
Does this help you:
```
diff --git a/django/test/utils.py b/django/test/utils.py
index cca757bda1..79874133d8 100644
--- a/django/test/utils.py
+++ b/django/test/utils.py
@@ -420,7 +420,7 @@ class override_settings(TestContextDecorator):
             setattr(override, key, new_value)
 
         if 'SECRET_KEY' in self.options:
-            override.SECRET_KEYS = [self.options['SECRET_KEY']]
+            override.SECRET_KEYS = [self.options['SECRET_KEY']] if self.options['SECRET_KEY'] else []
 
         if 'SECRET_KEYS' in self.options:
             override.SECRET_KEY = self.options['SECRET_KEYS'][0]
```

Reply:
Your current code copies the empty (!) SECRET_KEY into SECRET_KEYS, making it a list ala `[""]`

Reply:
Fwiw the line below can also fail if `SECRET_KEYS` is set to an empty list; in this case you should set `SECRET_KEY` to empty as well I think? We really have to check if we can make this code nicer, it makes my head hurt :)

Reply:
Yes. We do. 😄 — Still at the making it work phase. First real look this morning so, plenty of time 🤪